src/Configurator/ModifyFileTrait.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Flex\Configurator;
+
+trait ModifyFileTrait
+{
+    protected function shouldWriteFile(string $file, bool $overwrite): bool
+    {
+        if (!file_exists($file)) {
+            return true;
+        }
+
+        if (!$overwrite) {
+            return false;
+        }
+
+        if (!$this->gitExists()) {
+            return (bool) $this->io->askConfirmation(sprintf('Cannot determine the state of the "%s" file, overwrite anyway? [y/N] ', $file), false);
+        }
+
+        exec('git status --short --ignored -- '.escapeshellarg($file).' 2>&1', $output, $status);
+
+        $filePath = null;
+
+        if (0 === $status) {
+            $basename = basename($file);
+
+            $filePath = isset($output[0]) && $output[0]
+                ? (false !== strpos($output[0], $basename, -\strlen($basename)) ? substr($output[0], 3) : $file)
+                : ''
+            ;
+        }
+
+        if (file_exists($filePath ?: $file)) {
```

Comment:
file_exists($file) would be enough, isn't it? if yes, let's kiss

Reply:
`$filePath` can be null if the status code is different than 0, in which case we simply fallback to _"If exists, confirm removal"_ instead of _"If has modifications, confirm removal"_.

I could change to `if ((0 !== $status && file_exists($file)) || file_exists($filePath)) ` though, this should produce the same, but I'm not sure this simplifies things 😕